src/vcpkg/spdx.cpp

```diff
@@ -0,0 +1,256 @@
+    const auto& cpgh = *scfl.source_control_file->core_paragraph;
+    const StringView abi = action.package_abi() ? StringView(*action.package_abi().get()) : "NONE";
+
+    Json::Object doc;
+    doc.insert("$schema", "https://raw.githubusercontent.com/spdx/spdx-spec/v2.2.1/schemas/spdx-schema.json");
+    doc.insert("spdxVersion", "SPDX-2.2");
+    doc.insert("dataLicense", "CC0-1.0");
+    doc.insert("SPDXID", "SPDXRef-DOCUMENT");
+    doc.insert("documentNamespace", std::move(document_namespace));
+    doc.insert("name", Strings::concat(action.spec.to_string(), '@', cpgh.to_version().to_string(), ' ', abi));
+    {
+        auto& cinfo = doc.insert("creationInfo", Json::Object());
+        auto& creators = cinfo.insert("creators", Json::Array());
+        creators.push_back(Strings::concat("Tool: vcpkg-", VCPKG_VERSION_AS_STRING));
+        cinfo.insert("created", std::move(created_time));
+    }
+
+    auto& rels = doc.insert("relationships", Json::Array());
+    auto& packages = doc.insert("packages", Json::Array());
+    {
+        auto& obj = packages.push_back(Json::Object());
+        obj.insert("name", action.spec.name());
+        obj.insert("SPDXID", "SPDXRef-port");
+        obj.insert("versionInfo", cpgh.to_version().to_string());
+        obj.insert("downloadLocation", scfl.registry_location.empty() ? noassert : scfl.registry_location);
+        if (!cpgh.homepage.empty())
+        {
+            obj.insert("homepage", cpgh.homepage);
+        }
+        obj.insert("licenseConcluded", conclude_license(cpgh.license.value_or("")));
+        obj.insert("licenseDeclared", noassert);
+        obj.insert("copyrightText", noassert);
+        if (!cpgh.summary.empty()) obj.insert("summary", Strings::join("\n", cpgh.summary));
+        if (!cpgh.description.empty()) obj.insert("description", Strings::join("\n", cpgh.description));
+        obj.insert("comment", "This is the port (recipe) consumed by vcpkg.");
+        {
+            auto& rel = rels.push_back(Json::Object());
+            rel.insert("spdxElementId", "SPDXRef-port");
+            rel.insert("relationshipType", "GENERATES");
+            rel.insert("relatedSpdxElement", "SPDXRef-binary");
+        }
+        for (size_t i = 0; i < relative_paths.size(); ++i)
+        {
+            auto& rel = rels.push_back(Json::Object());
+            rel.insert("spdxElementId", "SPDXRef-port");
+            rel.insert("relationshipType", "CONTAINS");
+            rel.insert("relatedSpdxElement", Strings::concat("SPDXRef-file-", i));
+        }
+    }
+    {
+        auto& obj = packages.push_back(Json::Object());
+        obj.insert("name", action.spec.to_string());
+        obj.insert("SPDXID", "SPDXRef-binary");
+        obj.insert("versionInfo", abi);
+        obj.insert("downloadLocation", "NONE");
+        obj.insert("licenseConcluded", conclude_license(cpgh.license.value_or("")));
+        obj.insert("licenseDeclared", noassert);
+        obj.insert("copyrightText", noassert);
```

Comment:
For now I wanted to get basic stuff in place and see what additional data is required after. For example, maybe we can use the SPDX license "go look in that file" instead of having to copy the entire text.